fadecandy_driver/CMakeLists.txt

```diff
@@ -1,14 +1,66 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.0.2)
 project(fadecandy_driver)
 
-find_package(catkin REQUIRED)
+add_compile_options(-std=c++11)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
 
-catkin_python_setup()
 
-catkin_package()
+##################
+##  Build setup ##
+##################
 
-catkin_install_python(
-    PROGRAMS
-    scripts/fadecandy_node
-    DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-    )
+find_package(catkin REQUIRED COMPONENTS
+  auto_lint
+  roscpp
+  fadecandy_msgs
+)
+
+catkin_package(
+  CATKIN_DEPENDS
+    roscpp
+    fadecandy_msgs
+  LIBRARIES ${PROJECT_NAME}
+)
+
+##############
+##   Build  ##
+##############
+
+include_directories(
+  ${catkin_INCLUDE_DIRS}
+)
+
+add_library(${PROJECT_NAME}
+  src/${PROJECT_NAME}.cpp
+)
+target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
+
+add_library(${PROJECT_NAME}_ros
+  src/ros/${PROJECT_NAME}_ros.cpp
+)
+target_link_libraries(${PROJECT_NAME}_ros ${PROJECT_NAME} usb-1.0)
```

Comment:
No dependency defined

Reply:
would you like to take provide me some help here ? 
the library name is libusb and it's in library usb-1.0 and this is how I use it  in my header #include <libusb-1.0/libusb.h>
But everything I didn't work when I put it  it in find_package() but it's not able to find it maybe I am doing something wrong but I am not getting how can I add it in my dependencies I mean should it be usb-1.0 or libusb or libusb-1.0 ?

Reply:
@reinzor  would you like to talk a look at the CmakeLists.txt file and review the dependency ?